test/fixtures/s3-fixture/src/main/java/fixture/s3/S3HttpHandler.java

```diff
@@ -245,31 +245,48 @@ public void handle(final HttpExchange exchange) throws IOException {
 
             } else if (Regex.simpleMatch("GET /" + path + "/*", request)) {
                 final BytesReference blob = blobs.get(requestComponents.uri());
-                if (blob != null) {
-                    final String range = exchange.getRequestHeaders().getFirst("Range");
-                    if (range == null) {
-                        exchange.getResponseHeaders().add("Content-Type", "application/octet-stream");
-                        exchange.sendResponseHeaders(RestStatus.OK.getStatus(), blob.length());
-                        blob.writeTo(exchange.getResponseBody());
-                    } else {
-                        final Matcher matcher = Pattern.compile("^bytes=([0-9]+)-([0-9]+)$").matcher(range);
-                        if (matcher.matches() == false) {
-                            throw new AssertionError("Bytes range does not match expected pattern: " + range);
-                        }
-
-                        final int start = Integer.parseInt(matcher.group(1));
-                        final int end = Integer.parseInt(matcher.group(2));
-
-                        final BytesReference rangeBlob = blob.slice(start, end + 1 - start);
-                        exchange.getResponseHeaders().add("Content-Type", "application/octet-stream");
-                        exchange.getResponseHeaders()
-                            .add("Content-Range", String.format(Locale.ROOT, "bytes %d-%d/%d", start, end, rangeBlob.length()));
-                        exchange.sendResponseHeaders(RestStatus.OK.getStatus(), rangeBlob.length());
-                        rangeBlob.writeTo(exchange.getResponseBody());
-                    }
-                } else {
+                if (blob == null) {
                     exchange.sendResponseHeaders(RestStatus.NOT_FOUND.getStatus(), -1);
+                    return;
+                }
+                final String range = exchange.getRequestHeaders().getFirst("Range");
+                if (range == null) {
+                    exchange.getResponseHeaders().add("Content-Type", "application/octet-stream");
+                    exchange.sendResponseHeaders(RestStatus.OK.getStatus(), blob.length());
+                    blob.writeTo(exchange.getResponseBody());
+                    return;
+                }
+
+                // S3 supports https://www.rfc-editor.org/rfc/rfc9110.html#name-range. The AWS SDK v1.x seems to always generate range
+                // requests with a header value like "Range: bytes=start-end" where both {@code start} and {@code end} are always defined
+                // (sometimes to very high value for {@code end}). It would be too tedious to fully support the RFC so S3HttpHandler only
+                // supports when both {@code start} and {@code end} are defined to match the SDK behavior.
+                final Matcher matcher = Pattern.compile("^bytes=([0-9]+)-([0-9]+)$").matcher(range);
```

Comment:
I think that this regex doesn't work for the open ended range? i.e. `bytes=1-`

Reply:
Yes, I haven't implemented fully the RFC but only the cases covered by our usage of the AWS SDK (which I think always appends and `end` offset)

Reply:
That's also why the HTTP handler throws an assertion error if one of the start or end offset is missing